tools/cxbe/Xbe.cpp

```diff
@@ -131,16 +131,16 @@ Xbe::Xbe(const char *x_szFilename)
             goto cleanup;
         }
 
+        // generate ascii title from certificate title name
         setlocale( LC_ALL, "English" );
-
         //wcstombs(m_szAsciiTitle, m_Certificate.wszTitleName, 40);
         char *c = m_szAsciiTitle;
         char *d = (char *) m_Certificate.wszTitleName;
         while (*d) {
            *c++ = *d++;
            d++;
         }
-
+        *c = '\0';
```

Comment:
This addition on it's own is fine, but the preceding loop is highly problematic. If the XBE contains 40 unicode characters for the title name, then there will be no zero-termination of `wszTitleName`. As the loop stubbornly tries to find a zero, it will end up looking at the first byte of the "alternate title IDs" table, which cxbe sets to zero, so it will terminate with `c` pointing at the byte after the buffer, which makes `*c = '\0';` an out of bounds write.

To fix this, the loop would need to check for the field length, too, and the size of `m_szAsciiTitle` needs to be increased to 41 bytes to include space for the zero-terminator.

Reply:
I have added a hack to the original commit which *should* fix this, although I did not test it.

I tried to keep changes simple as I don't feel like adding much more code to this. There's *more* buffer overruns elsewhere and the code doesn't even do character conversion yet.

Reply:
The hack is good enough for me.